### PR TITLE
Close index reader in tests

### PR DESCRIPTION
**Pull request by javanna**
Not closing the index reader in unit tests did not cause issues so far, but it becomes a problem since we use newSearcher more often which randomly sets an executor to the searcher. This is because the executor is shutdown as part of the index reader closing listener, and if the reader is not closed, the test leaks threads which causes test failures.

This is more evident in the lucene_snapshot branch as lucene 9.8 will introduce unconditional offloading of tasks to the executor when provided (including sequential execution). In main, there's many cases where even if we have an executor, it is never used because execution is sequential (single slice) hence it won't leak threads in that case.

This commit fixes the unit tests that use `LuceneTestCase#newSearcher` to create the searcher, which may randomize the executor and need the reader to be closed as that triggers the shutdown of the executor too.

### CODE REVIEW

**Review by ChrisHegarty — APPROVED** [review]
LGTM

File: test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java [hunk 1]
```diff
@@ -156,6 +158,7 @@
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.CyclicBarrier;
 import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
```
Comment (jpountz) [inline test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java:161]: I see new imports, but they don't seem used?

**Review by jpountz — APPROVED** [review]
Nice!

File: test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java (hunk 1 above)
Comment (javanna) [reply test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java:161]: ha :) that's the previous hack to trigger the issue. need to remove these. strange that checkstyle does not complain.

**Pull request merged**